src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -554,42 +562,54 @@ private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellation
 
             var startupExceptions = new List<Exception>();
 
+            IAsyncEnumerator<ConfigurationClient> clientEnumerator = null;
+
             try
             {
                 while (true)
                 {
-                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetAllClients();
-
-                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
+                    try
```

Comment:
why do you use try/finally here for dispose, but await using in Refresh?

Reply:
Got you, let me update it.